com.unity.render-pipelines.universal/Shaders/Particles/ParticlesLitInput.hlsl

```diff
@@ -59,6 +68,24 @@ half4 SampleAlbedo(float2 uv, float3 blendUv, half4 color, float4 particleColor,
     return albedo;
 }
 
+// Returns clear coat parameters
+// .x/.r == mask
+// .y/.g == smoothness
+half2 SampleClearCoat(float2 uv)
```

Comment:
just a guess, but this looks like the kind of code that will be the same in the non-particle code, and should be in a shared header? (if it was duplicated from non-particle shader code)

If my guess is wrong, disregard my comment ;)

Reply:
Yes, that is a copy. I agree with you. I need to get some clarity on that, because I remember placing that originally in input "header" file, but got some review comments about it.

Reply:
Looked into this and it realized that it depends on the constants defined in the file so that's why it probably wasn't moved.
I would perhaps leave it like this for the time being.

Reply:
I'm going to do a separate PR and move all of the sampling functions into one file.